pkg/generate/sort_test.go

```diff
@@ -0,0 +1,160 @@
+			&DeviceList{goodBD3N1, &DeviceList{goodBD4N1, nil}}}},
+			resource.MustParse("1Gi"), 4}, []v1alpha1.BlockDevice{goodBD1N1, goodBD2N1, goodBD3N1, goodBD4N1}, false},
+		{"four node LinkedList, three BD required", args{&DeviceList{goodBD1N1, &DeviceList{goodBD2N1,
+			&DeviceList{goodBD3N1, &DeviceList{goodBD4N1, nil}}}},
+			resource.MustParse("1Gi"), 3}, []v1alpha1.BlockDevice{goodBD1N1, goodBD2N1, goodBD3N1}, false},
+		{"five node LinkedList, five BD required", args{&DeviceList{goodBD1N1, &DeviceList{goodBD2N1,
+			&DeviceList{goodBD3N1, &DeviceList{goodBD4N1, &DeviceList{goodBD5N1, nil}}}}},
+			resource.MustParse("1Gi"), 5}, []v1alpha1.BlockDevice{goodBD1N1, goodBD2N1, goodBD3N1, goodBD4N1, goodBD5N1}, false},
+		{"six node LinkedList, four BD required", args{&DeviceList{goodBD1N1, &DeviceList{goodBD2N1,
+			&DeviceList{goodBD3N1, &DeviceList{goodBD4N1, &DeviceList{goodBD5N1, &DeviceList{goodBD6N1, nil}}}}}},
+			resource.MustParse("1Gi"), 4}, []v1alpha1.BlockDevice{goodBD1N1, goodBD2N1, goodBD3N1, goodBD4N1}, false},
+		{"six node LinkedList, five BD required", args{&DeviceList{goodBD1N1, &DeviceList{goodBD2N1,
+			&DeviceList{goodBD3N1, &DeviceList{goodBD4N1, &DeviceList{goodBD5N1, &DeviceList{goodBD6N1, nil}}}}}},
+			resource.MustParse("1Gi"), 5}, []v1alpha1.BlockDevice{goodBD1N1, goodBD2N1, goodBD3N1, goodBD4N1, goodBD5N1}, false},
+		{"six node LinkedList, six BD required", args{&DeviceList{goodBD1N1, &DeviceList{goodBD2N1,
+			&DeviceList{goodBD3N1, &DeviceList{goodBD4N1, &DeviceList{goodBD5N1, &DeviceList{goodBD6N1, nil}}}}}},
+			resource.MustParse("1Gi"), 6}, []v1alpha1.BlockDevice{goodBD1N1, goodBD2N1, goodBD3N1, goodBD4N1, goodBD5N1, goodBD6N1}, false},
+		{"six node LinkedList, two BD required of 1G", args{bdLinkedList(6, []int{1, 2, 3, 4, 5, 6}), resource.MustParse("1G"), 2}, nil, true},
+		{"six node LinkedList, two BD required of 1G", args{bdLinkedList(6, []int{1, 2, 3, 4, 6, 6}), resource.MustParse("1G"), 2},
+			[]v1alpha1.BlockDevice{bdGen(5, 6), bdGen(6, 6)}, false},
+		{"six node LinkedList, two BD required of 1G", args{bdLinkedList(6, []int{1, 2, 4, 4, 6, 6}), resource.MustParse("1G"), 2},
+			[]v1alpha1.BlockDevice{bdGen(3, 4), bdGen(4, 4)}, false},
+		{"six node LinkedList, three BD required of 1G", args{bdLinkedList(6, []int{1, 4, 4, 4, 6, 6}), resource.MustParse("1G"), 3},
+			[]v1alpha1.BlockDevice{bdGen(2, 4), bdGen(3, 4), bdGen(4, 4)}, false},
+		{"six node LinkedList, three BD required of 1G", args{bdLinkedList(6, []int{5, 10, 15, 20, 25, 30}), resource.MustParse("1G"), 3},
+			nil, true},
+		{"six node LinkedList, three BD required of 1G", args{bdLinkedList(6, []int{1, 1, 10, 20, 25, 30}), resource.MustParse("1G"), 2},
+			[]v1alpha1.BlockDevice{bdGen(1, 1), bdGen(2, 1)}, false},
+		{"six node LinkedList, three BD required of 6G", args{bdLinkedList(6, []int{5, 10, 10, 20, 25, 30}), resource.MustParse("1G"), 2},
+			[]v1alpha1.BlockDevice{bdGen(2, 10), bdGen(3, 10)}, false},
```

Comment:
Can we add a test case to pass invalid arguments to select function? i.e passing sizes in non-sorted order?

Reply:
it'll fail with an error, good suggestion, it makes sense to also have a sane test case which is actually a negative one

Reply:
this would fail, but it'd be a good candidate for testing, adding it

Reply:
added `six node LinkedList with unsorted BD sizes, two BD required of 1G`, this request fails even when the BDs are present because they don't come under the same window of size=2 together